zebra-consensus/src/block/subsidy/funding_streams.rs

```diff
@@ -124,20 +124,43 @@ pub fn funding_stream_address(
 ///
 /// [7.10]: https://zips.z.cash/protocol/protocol.pdf#fundingstreams
 pub fn check_script_form(lock_script: &Script, address: Address) -> bool {
-    let mut address_hash = address
+    // TODO: Verify P2SH multisig funding stream addresses (#5577).
+    //       As of NU5, the funding streams do not use multisig addresses,
+    //       so this is optional.
```

Comment:
This is not the case. All of the (currently-revealed) funding stream addresses are P2SH ([as indicated by their `t3` prefixes](https://github.com/zcash/zcash/blob/fadbb794f16bee26513eb9a1fda51ac0cfa096e1/src/chainparams.cpp#L171-L225)), and of the addresses with publicly-revealed `scriptSig` values, all of the ones I spot-checked are multisig addresses. In particular, the ZF's funding stream address (the `scriptSig` for which is visible in [this transaction](https://blockchair.com/zcash/transaction/866de6abfee9ad02c267dccfc45710d8847b66a72e3b458f120c94ca0b4bf83c?i=999)) is a 2-of-3 multisig address (with an additional storage of 4 bytes of data that I can:

```
// OP_0
00
// PUSHDATA(<SIGNATURE> || SIGHASH_ALL)
48
  3045022100a16d32df92e28624ae194fb4de9e8610aa1932e4e2b370fa7c0ca7dab62d34d102205cc4f9dac56d43e7fd9e3d70816e3aa7f72832774712193bfbc227fb5ffaa0bf
  01
// PUSHDATA(<SIGNATURE> || SIGHASH_ALL)
47
  304402202341788cde33e568cfc388acf36ae9b24c22883c7ad88822e54c8bbe07a4c93202207654bcaa7c8f14403c424c7d681b4c525f779aa314795066a825e921fd4525d4
  01
// PUSHDATA(multisig script)
4c 72
  52 // OP_2
  21 0352d50656796dd79e3e3385c29636d849b8705c9c6d9b86a1717adc38e4a567c7 // PUSHDATA(pubkey)
  21 02f39d0b69c3b8e06030ef9dc295cf8c848f51e398fe85414e33509afdc7f01fb3 // PUSHDATA(pubkey)
  21 027072ba2319ead2b0b387d69d36508f57b397da8d4ac76ea64676ee8a1356dae0 // PUSHDATA(pubkey)
  53 // OP_3
  af // OP_CHECKMULTISIGVERIFY
  04 d5f2a8b6 // PUSHDATA(0xb6a8f2d5)
  75 // OP_DROP
  74 // OP_DEPTH
  00 // OP_0
  87 // OP_EQUAL
```

Reply:
Thanks, I'll deal with this in ticket #5577.